### PR TITLE
gear: userspace permissions interface (types + pokes)

### LINKED ISSUE #177: implement new types and pokes for userspace permissions
- per the backend PRs from palfun and tinnus, we need to add new types and pokes for userspace permissions
  - these will be added to the gear/ dir

### PR DESCRIPTION
**Pull request by tomholford**
# Context

This is part of an ongoing effort to implement Userspace Permissions UI (https://github.com/tloncorp/landscape/issues/182).

The Typescript types are derived from `lull.hoon` in this PR:
https://github.com/urbit/urbit/pull/6493/files#diff-01fd10efd392903ad3f191b1caddb67dba7d57d11997894173665d20f0d42477

and `lib/perms.hoon` from here:
https://github.com/tloncorp/landscape/pull/164/files#diff-418361e9c7bf8b945e9e34d68b4a363ba4ea46e2c400be84df6bdab9d15664e2

The updates for Permissions are in two separate repos: `urbit/urbit` and `tloncorp/landscape`:

- `passport` lives in `urbit/urbit` due to it's usage in the upcoming `|okay` perms mgmt cli tool
- Treaty and Pike updates live in `tloncorp/landscape`

This resolves #177 

# Changes

- [x] add new permissions types (see: urbit/urbit/pull/6493)
- [ ] add new permissions scries (see: urbit/urbit/pull/6493)
- [ ] add new permissions pokes (see: urbit/urbit/pull/6493)

Fixes LAND-514

### CODE REVIEW

File: ui/src/gear/hood/types.ts [hunk 1]
```diff
@@ -161,20 +162,15 @@ export interface Vat {
 export interface Vats {
   [desk: string]: Vat;
 }
+
 /**
  * TODO: crisp one-liner describing a Pike
```
Comment (tomholford) [inline ui/src/gear/hood/types.ts:167]: The People clamor to know: What is a Pike?

File: ui/src/gear/hood/types.ts [hunk 2]
```diff
@@ -201,6 +203,18 @@ export interface Pike {
    * dead - app is not running
    */
   zest: "live" | "dead" | "held";
+  /**
+   *  {@link Weft}s associated with this Pike
+   */
+  wefts: Weft[];
+  /**
+   *  {@link Perm}s granted to this Pike
```
Comment (tomholford) [inline ui/src/gear/hood/types.ts:211]: Are these accurate descriptions for `perms.pike` and `lacks.pike`?

File: ui/src/gear/permissions/types.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ */
+type Dude = string;
+
+/**
+ * path - ship desk case spur
+ */
+type Spur = string;
+
+/**
+ * TODO
```
Comment (tomholford) [inline ui/src/gear/permissions/types.ts:12]: `burr` and `spar` are defined thusly:

```hoon
  +$  burr  [desk=(unit desk) =spur]
  +$  spar  [care=(unit term) burr]
```

What is a good label for them in this type description?

**Comment by tomholford** [comment]
@tinnus-napbus or @Fang- : could use your feedback on the inline comments at your convenience

File: ui/src/gear/hood/types.ts (hunk 1 above)
Comment (Fang-) [reply ui/src/gear/hood/types.ts:167]: "Update config & status for desk" or something along those lines?

File: ui/src/gear/hood/types.ts (hunk 2 above)
Comment (Fang-) [reply ui/src/gear/hood/types.ts:211]: Yes.

File: ui/src/gear/permissions/types.ts (hunk 3 above)
Comment (Fang-) [reply ui/src/gear/permissions/types.ts:12]: Something like:

burr: specific or any desk, at or under spur prefix
spar: burr with specific or any care

**Comment by tomholford** [comment]
Based on out of band conversation with ~tinnus-napbus, some changes are necessary for this interface. Closing in favor of #195

**Pull request closed without merging**